### PR TITLE
chore: add 'themeName' option for ElementUiResolver

### PR DESCRIPTION
**Pull request by neterm**

### CODE REVIEW

**Review by sxzz — APPROVED** [review]

File: src/core/resolvers/element-ui.ts [hunk 1]
```diff
@@ -16,24 +16,43 @@ export interface ElementUiResolverOptions {
    * exclude component name, if match do not resolve the name
    */
   exclude?: RegExp
+
+  /**
+   * theme name, like '@qzdata/element-theme-qzdata'
+   */
```
Comment (sxzz) [inline src/core/resolvers/element-ui.ts:22]: Add `@default element-ui/packages/theme-chalk`

File: src/core/resolvers/element-ui.ts [hunk 2]
```diff
@@ -16,24 +16,43 @@ export interface ElementUiResolverOptions {
    * exclude component name, if match do not resolve the name
    */
   exclude?: RegExp
+
+  /**
+   * theme name, like '@qzdata/element-theme-qzdata'
```
Comment (sxzz) [inline src/core/resolvers/element-ui.ts:21]:
```suggestion
   * theme name
```

File: src/core/resolvers/element-ui.ts [hunk 3]
```diff
@@ -16,24 +16,43 @@ export interface ElementUiResolverOptions {
...
+   */
+  themeName?: string
 }
 
 function getSideEffects(
   partialName: string,
   options: ElementUiResolverOptions,
 ): SideEffectsInfo | undefined {
-  const { importStyle = 'css' } = options
+  const { importStyle = 'css', themeName } = options
```
Comment (sxzz) [inline src/core/resolvers/element-ui.ts:30]:
```suggestion
  const { importStyle = 'css', themeName = 'element-ui/packages/theme-chalk' } = options
```

File: src/core/resolvers/element-ui.ts [hunk 4]
```diff
@@ -16,24 +16,43 @@ export interface ElementUiResolverOptions {
...
     return
 
   if (importStyle === 'sass') {
+    if (themeName) {
+      return [
+        `${themeName}/src/base.scss`,
+        `${themeName}/src/${partialName}.scss`,
+      ]
+    }
+
```
Comment (sxzz) [inline src/core/resolvers/element-ui.ts:42]: We can remove `if (themeName) {` check if there's a default value.

File: src/core/resolvers/element-ui.ts [hunk 5]
```diff
@@ -16,24 +16,43 @@ export interface ElementUiResolverOptions {
...
     return [
       'element-ui/packages/theme-chalk/src/base.scss',
       `element-ui/packages/theme-chalk/src/${partialName}.scss`,
     ]
   }
   else {
+    if (themeName) {
```
Comment (sxzz) [inline src/core/resolvers/element-ui.ts:49]: Same here

**Review by sxzz — CHANGES_REQUESTED** [review]

**Comment by antfu** [comment]
Close due to inactive - feel free to bring it back if anyone interested in continue it

**Pull request closed without merging**